src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -85,5 +156,251 @@ public override bool Execute()
 
             return true;
         }
+
+        private void AddMappingsForCustomSources(XElement pkgSrcMappingElement, XElement pkgSourcesElement, List<string> packagePatterns)
+        {
+            foreach (string sourceName in CustomSources)
+            {
+                if (null != pkgSourcesElement.Descendants().FirstOrDefault(e => e.Name == "add" && e.Attribute("key").Value == sourceName))
```

Comment:
It looks like this logic is duplicated a number of times.  Consider a lookup helper method.